evennia/contrib/actions/handlers.py

```diff
@@ -0,0 +1,752 @@
+        once the current actions sharing the same bodyparts are finished.
+        A value of "ignore" means that newly-input actions that use the same
+        bodyparts as ongoing actions will not be performed at all.
+        """
+        return self.owner.db.actions['new']
+
+
+    @new.setter
+    def new(self, value):
+        if value == "override" or value == "queue" or value == "ignore":
+            self.owner.db.actions['new'] = value
+        else:
+            raise ValueError("Attempting to set value of {0}".format(value) +
+                " to actions new setting for character {0};".fomat(self.owner) + 
+                " actions new setting must be either \"override\", \"queue\"" +
+                " or \"ignore\".") 
+
+
+    @property
+    def movespeed(self):
+        """        
+        A function that takes a Character (the owner) as an argument and
+        returns the character's movement speed.
+        """
+        return self.owner.db.actions['movespeed']
+
+
+    @movespeed.setter
+    def movespeed(self, value):
+        self.owner.db.actions['movespeed'] = value
+
+
+    @property
+    def movetype(self):
+        """
+        A string that describes the character's current movement mode.
+        It is used in automatic invocations of the movebps function
+        by MoveOut actions (described in actions.py)
+        """
+        return self.owner.db.actions['movetype']
+
+
+    @movetype.setter
+    def movetype(self, value):
+
+        # To prevent characters from moving in multiple directions
+        # at once, changing a character's movement type cancels that
+        # character's present MoveOut action and is impossible if
+        # the character is performing a MoveIn action.
+
+        if owner.actions.movebps and owner.actions.movetype:
+            bodyparts = owner.actions.movebps(owner.actions.movetype)
+        else:
+            self.owner.db.actions['movetype'] = value
+            return
+
+        actions = [x for x in self.owner.location.db.actions if
+            x['owner'] == self.owner and x['bodyparts'] == bodyparts]
+        if actions:
+            # Check whether an uncancellable action with the same bodyparts
+            # as the present movement mode (e.g. MoveIn) is being undertaken
+            # by the character.
+            if [x for x in actions if not x['cancellable']]:
+                self.owner.msg("The movement mode cannot be changed at " +
+                    "present.")
+                return
+
+            for action in actions:
+                # Cancel all actions that use the same bodyparts as the
+                # present movement mode (e.g. Moveout)
+
+                # The reason we are not selecting actions by key is that
+                # someone, somewhere may want to subclass MoveOut and MoveIn.
+                action['at_cancel'](action)            
+
+        self.owner.db.actions['movetype'] = value
+
+    @property
+    def movebps(self):
```

Comment:
How do you feel about "bpsformove"? It's shorter (more python-themed) and still gets the point across. Alternatively, maybe "bpmovemap" or "bp_move_map". If you prefer "bodypart_movement_map", I'll go with that in the end. What are your thoughts on this?